internal/ethapi/api.go

```diff
@@ -1238,10 +1238,11 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 		balance := state.GetBalance(*args.From) // from can't be nil
 		available := new(big.Int).Set(balance)
 		if args.Value != nil {
-			if args.Value.ToInt().Cmp(available) >= 0 {
-				return 0, core.ErrInsufficientFundsForTransfer
+			value := args.Value.ToInt()
+			if value.Cmp(available) >= 0 {
```

Comment:
Hm, isn't it odd that the comparison (now and before) uses `>=`, as opposed to just `>` ? Why error on equality?

Reply:
IMHO, as `feeCap` is not zero, we need to consider the gasFee, so `=` is not enough, maybe I need to adjust the error message.

Reply:
Changed the error from `core.ErrInsufficientFundsForTransfer` to `core.ErrInsufficient`(the later contains the `gas*price` part)

Reply:
Why not `ErrInsufficientFundsForTransfer`? This clause is specifically comparing the `value` against the `balance`.

Reply:
As `ErrInsufficientFundsForTransfer` only considered the transferred value's part, not including the gasFee, and here as the `feeCap` is not zero, we need to add the feeCap. So `ErrInsufficientFunds` probably more accurate, because his error log is `insufficient funds for gas * price + value`